### ISSUE TITLE
create-react-app React performance with SVG as ReactComponent and eslint with React.memo missing display name

### ISSUE THREAD

**Issue by frederikhors** [yannickcr/eslint-plugin-react#2254]
I'm using this component in a `create-react-app` app:

```js
import { ReactComponent as ProfileIcon } from "./icons/profile.svg";

...
render(){
  <ProfileIcon {...props}/>
}
```

Using also `why-did-you-render` (https://github.com/welldone-software/why-did-you-render) I got this warning:

```log
SvgProfileIcon
whyDidYouRender.min.js:1191 {SvgProfileIcon: ƒ} "Re-rendered because the props object itself changed but it's values are all equal." "This could of been avoided by making the component pure, or by preventing it's father from re-rendering." "more info at http://bit.ly/wdyr02"
whyDidYouRender.min.js:1191 prev props: {svgRef: null, className: "icon", height: "24", width: "24"} !== {svgRef: null, className: "icon", height: "24", width: "24"} :next props
```

So I made a custom `PureComponent` like this:

```js
import React from "react";

export default WrappedComponent =>
  React.memo(props => <WrappedComponent {...props} />, () => true);
```

**FIRST QUESTION**: Is this performances-correct?

I'm using it like this:

```js
import { ReactComponent as ProfileIcon } from "./icons/profile.svg";

import PureComponent from "./PureComponent";

const PureProfileIcon = PureComponent(ProfileIcon);

...
render(){
  <PureProfileIcon {...props}/>
}
```

**SECOND QUESTION**: Can I avoid this component at all using React.memo (or something else) differently?

Now eslint is complaining about:

```error
Component definition is missing display name eslint(react/display-name)
```

**THIRD QUESTION**: How can I fix this?

**Comment by ljharb** [comment]
To fix the lint warning, don't use an arrow function - use a named function.

**Comment by frederikhors** [comment]
Thanks @ljharb, but how?

**Comment by ljharb** [comment]
```jsx
 React.memo(function SomeName(props) { return <WrappedComponent {...props} />; }, () => true);
```
